src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4encoder/HaivisionX4EncoderCommunicator.java

```diff
@@ -0,0 +1,894 @@
+							String.valueOf(videoMap.get(Integer.valueOf(state))));
+				}
+			} else {
+				stats.put(videoName + HaivisionConstant.SPACE + HaivisionConstant.STATISTICS + HaivisionConstant.HASH + videoMetric.getName(), videoStatistic.getValueByMetric(videoMetric));
+			}
+		}
+	}
+
+	/**
+	 * Add output data to statistics property
+	 *
+	 * @param stats list statistics property
+	 */
+	private void populateOutputData(Map<String, String> stats) {
+		if (isAdapterFilter) {
+			if (!outputForPortAndStatusList.isEmpty()) {
+				for (OutputResponse outputResponses : outputForPortAndStatusList) {
+					addOutputStreamDataStatisticsToStatisticsProperty(stats, outputResponses);
+				}
+			} else {
+				for (OutputResponse outputResponses : outputStatisticsList) {
+					addOutputStreamDataStatisticsToStatisticsProperty(stats, outputResponses);
+				}
+			}
+		} else {
+			for (OutputResponse outputResponses : outputResponseList) {
+				addOutputStreamDataStatisticsToStatisticsProperty(stats, outputResponses);
+			}
+		}
+	}
+
+	/**
+	 * Add output stream data to statistics property
+	 *
+	 * @param stats list statistics property
+	 * @param outputResponseList list of output response
+	 */
+	private void addOutputStreamDataStatisticsToStatisticsProperty(Map<String, String> stats, OutputResponse outputResponseList) {
+		String name = convertStreamNameUnescapeHtml3(outputResponseList.getName());
+		OutputStatistic outputStatistic = outputResponseList.getOutputStatistic();
+		Map<Integer, String> outputMap = OutputStateDropdown.getNameToValueMap();
+		String state = outputResponseList.getState();
+		for (OutputMonitoringMetric outputStreamMetric : OutputMonitoringMetric.values()) {
+			if (OutputMonitoringMetric.UPTIME.equals(outputStreamMetric) || OutputMonitoringMetric.OCCURRED.equals(outputStreamMetric)) {
+				stats.put(name + HaivisionConstant.SPACE + HaivisionConstant.STATISTICS + HaivisionConstant.HASH + outputStreamMetric.getName(),
+						formatTimeData(outputStatistic.getValueByMetric(outputStreamMetric)));
+				continue;
+			}
+			if (OutputMonitoringMetric.STATE.equals(outputStreamMetric)) {
+				if (HaivisionConstant.NONE.equals(state)) {
+					stats.put(String.format(HaivisionConstant.FORMAT, name + HaivisionConstant.SPACE + HaivisionConstant.STATISTICS, outputStreamMetric.getName()), state);
+				} else {
+					stats.put(String.format(HaivisionConstant.FORMAT, name + HaivisionConstant.SPACE + HaivisionConstant.STATISTICS, outputStreamMetric.getName()),
+							String.valueOf(outputMap.get(Integer.valueOf(state))));
+				}
+			} else {
+				stats.put(name + HaivisionConstant.SPACE + HaivisionConstant.STATISTICS + HaivisionConstant.HASH + outputStreamMetric.getName(), outputStatistic.getValueByMetric(outputStreamMetric));
+			}
+		}
+	}
+
+	/**
+	 * Retrieve audio encoder
+	 */
+	private void retrieveAudioEncoder() {
+		try {
+			String responseData = doGet(HaivisionStatisticsUtil.getMonitorURL(HaivisionURL.AUDIO_ENCODER));
```

Comment:
would it work if you used DTO in doGet directly (with maybe a wrapper that seems to be necessary here, like {"data": DTO}), instead of mapping it separately?